src/Microsoft.Data/DataFrameTable.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A DataFrameTable is just a container that holds a number of DataFrameColumns. It mainly acts as a convenient store to allow DataFrame to implement its algorithms
+    /// </summary>
+    internal class DataFrameTable
+    {
+        private IList<BaseDataFrameColumn> _columns;
+
+        private List<string> _columnNames = new List<string>();
+        private Dictionary<string, int> _columnNameToIndexDictionary = new Dictionary<string, int>();
```

Comment:
When constructing a dictionary with string keys, it's nice to always specify a comparer. Since half the time there isn't one, you intended case insensitive, or you intended culture invariant. It makes it explicit

Reply:
I'm guessing you want STringComparer.CurrentCulture or Ordinal, without IgnoreCase, ?